source/ui/AMMotorGroupView.cpp

```diff
@@ -17,520 +17,1042 @@ GNU General Public License for more details.
+
+AMMotorGroupView::AMMotorGroupView(AMMotorGroup* motorGroup,
+								   ViewMode viewMode,
+								   QWidget* parent)
+	: QWidget(parent)
 {
-	if (viewMode_ == Exclusive){
+	motorGroup_ = motorGroup;
 
-		menu->addAction("Switch to Multiple View");
+	if(!motorGroup_) {
+		return;
 	}
 
-	else if (viewMode_ == Multiple){
+	if(viewMode == NormalView) {
+		setupNormalUi();
+	} else {
+		setupCompactUi();
+	}
 
-		menu->addAction("Switch to Exclusive View");
-		menu->addSeparator();
+}
 
-		for (int i = 0, size = motorGroupViews_.size(); i < size; i++){
+AMMotorGroupObject * AMMotorGroupView::selectedGroupObject() const
+{
+	if(currentSelectedGroupObjectName_.isEmpty()) {
+		return 0;
+	}
 
-			QAction *action = new QAction(motorGroupViews_.keys().at(i), menu);
-			action->setCheckable(true);
-			action->setChecked(motorGroupViews_.values().at(i)->isVisible());
-			menu->addAction(action);
-		}
+	return motorGroup_->motorGroupObject(currentSelectedGroupObjectName_);
+}
+
+void AMMotorGroupView::onGroupObjectMotionStatusAltered(const QString& groupObjectName)
+{
+	AMMotorGroupObject* motorGroupObject = motorGroup_->motorGroupObject(groupObjectName);
+	QIcon statusIcon;
+	if(motorGroupObject->isMoving()) {
+		statusIcon = QIcon(":/32x32/greenLEDOn.png").pixmap(25);
+	} else {
+		statusIcon = QIcon(":/32x32/greenLEDOff.png").pixmap(25);
+	}
+
+	int tabIndex = motorGroupTabMap_.value(groupObjectName, -1);
+
+	if(tabIndex >= 0 ) {
+		groupObjectTabs_->setTabIcon(tabIndex, statusIcon);
 	}
 }
 
-QString AMMotorGroupView::currentMotorGroupObjectName() const
+void AMMotorGroupView::onGroupComboBoxIndexChanged(int index)
 {
-	return viewMode_ == Exclusive ? currentMotorGroupObjectView_->motorGroupObject()->name() : "";
+	groupObjectStack_->setCurrentIndex(index);
+
+	QString nameOfSelectedGroup =
+			groupObjectSelector_->itemText(index);
+
+	currentSelectedGroupObjectName_ = nameOfSelectedGroup;
+	emit currentMotorGroupObjectViewChanged(nameOfSelectedGroup);
 }
 
-AMMotorGroupObjectView *AMMotorGroupView::currentMotorGroupObjectView() const
+void AMMotorGroupView::onGroupTabViewIndexChanged(int index)
 {
-	return viewMode_ == Exclusive ? currentMotorGroupObjectView_ : 0;
+	QString nameOfSelectedGroup =
+			motorGroupTabMap_.key(index);
+
+	currentSelectedGroupObjectName_ = nameOfSelectedGroup;
+	emit currentMotorGroupObjectViewChanged(nameOfSelectedGroup);
 }
 
-QStringList AMMotorGroupView::visibleMotorGroupObjectNames() const
+void AMMotorGroupView::setupCompactUi()
 {
-	if (viewMode_ == Multiple){
+	// Zero widgets we're not going to use:
+	groupObjectTabs_ = 0;
+	groupMotionStatusMapper_ = 0;
 
-		QStringList visibleNames;
+	// Set up the stuff we are going to use.
+	QVBoxLayout* mainLayout = new QVBoxLayout();
 
-		foreach (AMMotorGroupObjectView *view, motorGroupViews_.values())
-			if (view->isVisible())
-				visibleNames << view->motorGroupObject()->name();
+	groupObjectSelector_ = new QComboBox();
+	groupObjectStack_ = new QStackedWidget();
```

Comment:
I like this as a solution rather than my old one where I just hid them.  Nice change up.